src/ome_zarr_models/_rfc5_transforms/multiscales.py

```diff
@@ -0,0 +1,247 @@
+from __future__ import annotations
+
+from typing import Self
+
+from pydantic import (
+    BaseModel,
+    Field,
+    JsonValue,
+    field_validator,
+    model_validator,
+)
+
+from ome_zarr_models._rfc5_transforms.coordinate_transformations import (
+    CoordinateSystem,
+    CoordinateTransformationType,
+    Scale,
+    Translation,
+)
+from ome_zarr_models.base import BaseAttrs
+from ome_zarr_models.common.validation import check_length
+
+__all__ = ["Dataset"]
+
+
+class Dataset(BaseAttrs):
+    """
+    An element of Multiscale.datasets.
+    """
+
+    path: str
+    # TODO: suggest John not to have a tuple for the coordinate transformations of a
+    #  dataset, since it's either a single scale or a single sequence (with a scale
+    #  and a translation)
+    coordinateTransformations: tuple[CoordinateTransformationType, ...] = Field(
+        ..., min_length=1, max_length=1
+    )
+
+    # the before validation is used to simplify the error messages
+    @field_validator("coordinateTransformations", mode="before")
+    def _ensure_scale_translation(
+        transforms_obj: object,
+    ) -> object:
+        """
+        Ensures that
+        - a single transformation is present
+        - such transformation is a scale
+        - if such transformation is a sequence, ensure that its length is 2 and that
+          the first transformation is a scale and the second a translation
+        """
+
+        class Transforms(BaseModel):
```

Comment:
why are we defining a class inside a validator function?

Reply:
My comment on the existing code is

```python
        # This is used as a before validator - to help use, we use pydantic to first
        # cast the input (which can in general anything) into a set of transformations.
        # Then we check the transformations are valid.
        #
        # This is a bit convoluted, but we do it because the default pydantic error
        # messages are a mess otherwise
```

I'd suggest copy/pasting/modifying that here, to avoid confusion because defining the class here is a bit weird.